device/core/src/internal_client.ts

```diff
@@ -381,23 +445,26 @@ export abstract class InternalClient extends EventEmitter {
    * If no component is specified, then the default component is assumed.
    * This method is only intended for use with Azure IoT Plug and Play.
    *
-   * @param {string} commandName   The name of the command.
-   * @param {string} componentName The name of the component that corresponds with the command.
-   * @param {function} callback    The callback that is called each time a command request for this command is received.
+   * @param {string}   commandName   The name of the command.
+   * @param {string}   componentName The name of the component that corresponds with the command.
+   * @param {function} callback      The callback that is called each time a command request for this command is received.
    */
   onCommand(commandName: string, callback: (request: CommandRequest, response: CommandResponse) => void): void;
-  onCommand(commandName: string, componentName: string, callback: (request: CommandRequest, response: CommandResponse) => void): void;
-  onCommand(commandName: string, callbackOrComponent: ((request: CommandRequest, response: CommandResponse) => void) | string, callback?: (request: CommandRequest, response: CommandResponse) => void): void {
+  onCommand(componentName: string, commandName: string, callback: (request: CommandRequest, response: CommandResponse) => void): void;
+  onCommand(commandOrComponent: string, callbackOrCommand: ((request: CommandRequest, response: CommandResponse) => void) | string, callback?: (request: CommandRequest, response: CommandResponse) => void): void {
+    let commandName: string;
```

Comment:
Maybe it's the wine, but it looks like you're obscuring the parameter with the local.

Reply:
I mean are we doing this other places?

Reply:
See the my other reply to a similar comment.